Common/Utility/DificultyUtils.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Terraria;
+using Terraria.ModLoader;
+
+namespace TheBereftSouls.Common.Utility
+{
+    public class DificultyUtils : ModSystem
+    {
+        public static bool Revengeance = false;
+        public static bool Death = false;
+        public static bool EternityMode = false;
+        public static bool MasochistMode = false;
+        public static bool LegendaryMode = false;
+
+        public override void PostUpdateWorld()
+        {
+            CheckDificulty();
+            base.PostUpdateWorld();
+        }
+        public void CheckDificulty()
+        {
+            if(TheBereftSouls.CalamityMod != null && TheBereftSouls.FargosSoulMod != null)
+            {
+                CheckCalamity();
+                CheckFargos();
+
+            }
+            else if (TheBereftSouls.CalamityMod != null)
+            {
+                CheckCalamity();
+            }
+            else if (TheBereftSouls.FargosSoulMod != null)
+            {
+                CheckFargos();
+            }
+            LegendaryMode = Main.masterMode && Main.getGoodWorld;
+        }
+
+        [JITWhenModsEnabled("CalamityMod")]
+        public void CheckCalamity()
+        {
+            if (ModLoader.TryGetMod("CalamityMod", out Mod calamityMod))
+            {
+                Revengeance = (bool)calamityMod.Call("GetDifficultyActive", "revengeance");
+                Death = (bool)calamityMod.Call("GetDifficultyActive", "death");
+            }
+        }
```

Comment:
This is tricky, `the tModLoader devs recommend to JIT the whole class`[^1] with the mod and set a Load instance like so

- [Load() example with MagicStorage](https://github.com/JavidPack/ItemChecklist/blob/ccfcfc1e881aa099ad587dcd3bae7e43a5f48254/MagicStorageIntegration.cs#L49-L56)

> [!IMPORTANT]
> Read the comments from that link as they're very descriptive AND important. You can even read the whole source file documentation as it's also good to know.

[^1]: [tModLoader expert cross mod content recommendations](https://github.com/tModLoader/tModLoader/wiki/Expert-Cross-Mod-Content#recommendations)